### PR TITLE
tests: Fix tests crashing on Intel Mesa

### PR DESCRIPTION
**Pull request by spencer-lunarg**
reasons for each commented on

### CODE REVIEW

File: layers/core_checks/cc_pipeline_graphics.cpp [hunk 1]
```diff
@@ -987,8 +987,7 @@ bool CoreChecks::ValidateGraphicsPipelineTessellationState(const vvl::Pipeline &
     bool skip = false;
 
     if (pipeline.OwnsSubState(pipeline.pre_raster_state) &&
-        (pipeline.create_info_shaders & VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT) &&
-        (pipeline.create_info_shaders & VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT)) {
+        (pipeline.create_info_shaders & VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT)) {
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_pipeline_graphics.cpp:990]: VU only says "control" shader, was not catching GPL test bug

File: tests/unit/best_practices.cpp [hunk 2]
```diff
@@ -799,7 +799,7 @@ TEST_F(VkBestPracticesLayerTest, ClearAttachmentsAfterLoadSecondary) {
     auto image_view = image.CreateView();
 
     RenderPassSingleSubpass rp(*this);
-    rp.AddAttachmentDescription(VK_FORMAT_R8G8B8A8_UNORM, VK_IMAGE_LAYOUT_UNDEFINED, VK_IMAGE_LAYOUT_UNDEFINED,
```
Comment (spencer-lunarg) [inline tests/unit/best_practices.cpp:802]: I need to do a follow-up why this wasn't caught, but you can't final layout into UNDEFINED

File: tests/unit/dynamic_state.cpp [hunk 3]
```diff
@@ -5330,6 +5330,7 @@ TEST_F(NegativeDynamicState, DynamicSampleLocationsEnable) {
     CreatePipelineHelper pipe(*this);
     pipe.InitState();
     pipe.AddDynamicState(VK_DYNAMIC_STATE_SAMPLE_LOCATIONS_ENABLE_EXT);
+    pipe.AddDynamicState(VK_DYNAMIC_STATE_SAMPLE_LOCATIONS_EXT);
```
Comment (spencer-lunarg) [inline tests/unit/dynamic_state.cpp:5333]: https://gitlab.khronos.org/vulkan/vulkan/-/issues/3732

File: tests/unit/wsi.cpp [hunk 4]
```diff
@@ -2733,13 +2733,9 @@ TEST_F(NegativeWsi, CreatingWaylandSurface) {
     GTEST_SKIP() << "test not supported on platform";
 #else
     AddSurfaceExtension();
-
+    AddRequiredExtensions(VK_KHR_WAYLAND_SURFACE_EXTENSION_NAME);
```
Comment (spencer-lunarg) [inline tests/unit/wsi.cpp:2736]: not a bug, but realize the WSI tests were the old way (when debugging a test)

File: tests/unit/best_practices.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply tests/unit/best_practices.cpp:802]: `VUID-VkAttachmentDescription-finalLayout-00843`

Comment (spencer-lunarg) [reply tests/unit/best_practices.cpp:802]: awww, because it was best practice layer and wasn't triggering normal errors

**Review by arno-lunarg — APPROVED** [review]

**Pull request merged**